packages/autocomplete-js/src/__tests__/autocomplete.test.ts

```diff
@@ -156,6 +155,162 @@ describe('autocomplete-js', () => {
     `);
   });
 
+  test('renders empty template on empty results', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      id: 'autocomplete',
+      container,
+      panelContainer,
+      debug: true,
+      openOnFocus: true,
+      shouldPanelShow: () => true,
+      getSources() {
+        return [
+          {
+            getItems() {
+              return [];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+              empty() {
+                return 'No results template';
+              },
+            },
+          },
+        ];
+      },
+    });
+
+    const input = container.querySelector<HTMLInputElement>('.aa-Input');
+
+    fireEvent.input(input, {
+      target: { value: 'aasdjfaisdf' },
+    });
+    input.focus();
+
+    await waitFor(() => {
+      expect(
+        panelContainer.querySelector<HTMLElement>('.aa-Panel')
+      ).toBeInTheDocument();
+    });
+
+    expect(
+      panelContainer.querySelector<HTMLElement>('.aa-Panel').textContent
+    ).toEqual('No results template');
+  });
+
+  test('renders renderEmpty method on empty results', async () => {
+    const container = document.createElement('div');
+    const panelContainer = document.createElement('div');
+
+    document.body.appendChild(panelContainer);
+    autocomplete<{ label: string }>({
+      id: 'autocomplete',
+      container,
+      panelContainer,
+      debug: true,
+      openOnFocus: true,
+      shouldPanelShow: () => true,
+      getSources() {
+        return [
+          {
+            getItems() {
+              return [];
+            },
+            templates: {
+              item({ item }) {
+                return item.label;
+              },
+            },
+          },
+        ];
+      },
+      renderEmpty({ root }) {
```

Comment:
We can store this in a variable and wrap it in `jest.fn()` to make sure that `renderEmpty` was also called with the correct arguments.